src/index.ts

```diff
@@ -30,40 +30,104 @@
 
 import Ui from './ui';
 import Uploader from './uploader';
 
 import { IconAddBorder, IconStretch, IconAddBackground, IconPicture } from '@codexteam/icons';
+import { TunesConfig } from './types/types';
+
+/**
+ * ImageToolData interface representing the input and output data format for the image tool.
+ */
+export interface ImageToolData {
+  /**
+   * Caption for the image.
+   */
+  caption: string;
+  /**
+   * Flag indicating whether the image has a border.
+   */
+  withBorder: boolean;
+  /**
+   * Flag indicating whether the image has a background.
+   */
+  withBackground: boolean;
+  /**
+   * Flag indicating whether the image is stretched.
+   */
+  stretched: boolean;
+  /**
+   * Object containing the URL of the image file.
+   */
+  file: {
+    url: string;
+  };
+}
 
 /**
- * @typedef {object} ImageConfig
+ *
  * @description Config supported by Tool
- * @property {object} endpoints - upload endpoints
- * @property {string} endpoints.byFile - upload by file
- * @property {string} endpoints.byUrl - upload by URL
- * @property {string} field - field name for uploaded image
- * @property {string} types - available mime-types
- * @property {string} captionPlaceholder - placeholder for Caption field
- * @property {object} additionalRequestData - any data to send with requests
- * @property {object} additionalRequestHeaders - allows to pass custom headers with Request
- * @property {string} buttonContent - overrides for Select File button
- * @property {object} [uploader] - optional custom uploader
- * @property {function(File): Promise.<UploadResponseFormat>} [uploader.uploadByFile] - method that upload image by File
- * @property {function(string): Promise.<UploadResponseFormat>} [uploader.uploadByUrl] - method that upload image by URL
  */
+export interface ImageToolConfig {
+  /**
+   * Endpoints for upload, whether using file or URL.
+   */
+  endpoints: {
+    /**
+     * Endpoint for file upload.
+     */
+    byFile?: string;
+    /**
+     * Endpoints for URL upload.
+     */
+    byUrl?: string;
+  };
+  /**
+   * Field name for the uploaded image.
+   */
+  field?: string;
+  /**
+   * Allowed mime-types for the uploaded image.
+   */
+  types?: string;
+  /**
+   * Placeholder text for the caption field.
+   */
+  captionPlaceholder?: string;
+  /**
+   * Additional data to send with requests.
+   */
+  additionalRequestData?: object;
+  /**
+   * Additional headers to send with requests.
+   */
+  additionalRequestHeaders?: object;
+  /**
+   * Custom content for the select file button.
+   */
+  buttonContent?: string;
+  /**
+   * Optional custom uploader.
+   */
+  uploader?: {
+    /**
+     * Method to upload an image by file.
+     */
+    uploadByFile?: (file: Blob) => Promise<UploadResponseFormat>;
+    /**
+     * Method to upload an image by URL.
+     */
+    uploadByUrl?: (url: string) => Promise<UploadResponseFormat>;
+  };
+  /**
+   * Additional actions for the tool.
+   */
+  actions?: Array<TunesConfig>;
```

Comment:
maybe rename TunesConfig to ActionConfig? It will be more straightforward 
```suggestion
  actions?: ActionConfig[];
```